src/tdsops.f90

```diff
@@ -0,0 +1,682 @@
+module m_tdsops
+   use m_common, only: dp, pi
+
+   implicit none
+
+   type :: tdsops_t
+      !! Tridiagonal Solver Operators class.
+      !! Operator arrays are preprocessed in this class based on the arguments
+      !! provided. dist_fw and dist_bw are used in the first phase of the
+      !! distributed tridiagonal solver algorithm. dist_sa and dist_sc are used
+      !! in the final substutition phase. See the kernels_dist.f90 files in the
+      !! relevant backend folders.
+      !! coeff arrays define the specific rules of building the RHS
+      !! corresponding to the tridiagonal system to be solved, and used only in
+      !! the first phase of the distributed algorithm when building the RHS.
+      !! If a boundary condition is defined then coeffs_t and coeffs_b differ
+      !! from coeffs array and define the RHS rule for the first and last 4
+      !! entries in the tridiagonal system (n_halo = 4).
+      !!
+      !! This class does not know about the current rank or its relative
+      !! location among other ranks. All the operator arrays here are used when
+      !! executing a distributed tridiagonal solver phase one or two.
+      real(dp), allocatable, dimension(:) :: dist_fw, dist_bw, dist_sa, dist_sc
+      real(dp), allocatable :: coeffs(:), coeffs_t(:, :), coeffs_b(:, :)
+      integer :: n, n_halo
+      real(dp) :: alpha, delta
```

Comment:
Is delta the size between nodes ? If yes, it could be removed and integrated in the coefficients ? What about cases with a stretched mesh ?

Reply:
I think you're right, we don't need to access delta from the tdsops_t class, so it should be an input and not a member of the type itself. I'll remove if I can't find a use case for this.

Reply:
For the stretched mesh I think we need one more allocatable array like dist_st. We can add this when we need. There'll need to be a relevant change in one of the kernels_dist.f90 functions as well, I think only in the second phase function.

Reply:
Maybe we could consider that the derivative (on a streched mesh) is another layer sitting on top of this one ?

Reply:
This could be an elegant way to deal with both stretched meshes and specific cases such as nz=1.

Reply:
I think for performance it is better if we fuse the stretching operation with the second phase of the distributed algorithm. And if there is no stretching we will be multiplying the result array with an array of ones. Implementing this as a layer on top would require a dedicated function call that would add some extra cost due to moving data back and forth in memory